test/dynamo/test_skip_non_tensor.py

```diff
@@ -106,6 +159,27 @@ def __len__(self):
 
         assert counter.op_count == 0
 
+    def test_do_not_skip_side_effects(self):
+        # https://github.com/pytorch/pytorch/issues/110765
+        global _variable, _variable_2
+
+        for mode in (0, 1, 2, 3, 4):
+            _variable = 0
+            _variable_2 = 0
+
+            mod = MyModule(mode=mode)
+            model = torch.compile(mod, backend="eager")
```

Comment:
Could you use dynamo.optimize and perhaps nopython=True for majority of the cases?